algorithms/unit_tests/TestStdAlgorithmsTeamUniqueCopy.cpp

```diff
@@ -0,0 +1,180 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestStdAlgorithmsCommon.hpp>
+
+namespace Test {
+namespace stdalgos {
+namespace TeamUniqueCopy {
+
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType>
+struct CustomPredicate {
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(ValueType a, ValueType b) const { return a == b; }
+};
+
+template <class SourceViewType, class DestViewType, class DistancesViewType>
+struct TestFunctorA {
+  SourceViewType m_sourceView;
+  DestViewType m_destView;
+  DistancesViewType m_distancesView;
+  int m_apiPick;
+
+  TestFunctorA(const SourceViewType sourceView, const DestViewType destView,
+               const DistancesViewType distancesView, int apiPick)
+      : m_sourceView(sourceView),
+        m_destView(destView),
+        m_distancesView(distancesView),
+        m_apiPick(apiPick) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+    auto myRowViewFrom =
+        Kokkos::subview(m_sourceView, myRowIndex, Kokkos::ALL());
+    auto myRowViewDest = Kokkos::subview(m_destView, myRowIndex, Kokkos::ALL());
+
+    if (m_apiPick == 0) {
+      auto it =
+          KE::unique_copy(member, KE::begin(myRowViewFrom),
+                          KE::end(myRowViewFrom), KE::begin(myRowViewDest));
+
+      Kokkos::single(Kokkos::PerTeam(member), [=, *this]() {
+        m_distancesView(myRowIndex) =
+            KE::distance(KE::begin(myRowViewDest), it);
+      });
```

Comment:
this test doesn't trigger if any thread other than thread 0 will give the wrong answer.

Reply:
improved tests for this , let me know if the new solutions works for you